test/test_encryption.py

```diff
@@ -2202,6 +2202,83 @@ def test_05_roundtrip_encrypted_unindexed(self):
         self.assertEqual(decrypted, val)
 
 
+# https://github.com/mongodb/specifications/blob/072601/source/client-side-encryption/tests/README.rst#rewrap
+class TestRewrapWithSeparateClientEncryption(EncryptionIntegrationTest):
+
+    MASTER_KEYS: Mapping[str, Mapping[str, Any]] = {
+        "aws": {
+            "region": "us-east-1",
+            "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+        },
+        "azure": {
+            "keyVaultEndpoint": "key-vault-csfle.vault.azure.net",
+            "keyName": "key-name-csfle",
+        },
+        "gcp": {
+            "projectId": "devprod-drivers",
+            "location": "global",
+            "keyRing": "key-ring-csfle",
+            "keyName": "key-name-csfle",
+        },
+        "kmip": {},
+    }
+
+    def test_rewrap(self):
+        for src_provider in self.MASTER_KEYS:
+            for dst_provider in self.MASTER_KEYS:
+                with self.subTest(src_provider=src_provider, dst_provider=dst_provider):
+                    self.run_test(src_provider, dst_provider)
```

Comment:
The test says we still need to test the "local" provider:
> For "local", do not set a masterKey document.

I think that means we need to add
```
...
        "kmip": {},
        "local": None,
    }
```

Reply:
They used `local: {}` in the [C driver](https://github.com/mongodb/mongo-c-driver/pull/1085/files), I'll go with that.